pkgs/development/python-modules/cryptography/default.nix

```
@@ -49,11 +51,12 @@ buildPythonPackage rec {
   nativeBuildInputs = lib.optionals (!isPyPy) [
     cffi
   ] ++ [
-    rustPlatform.cargoSetupHook
-    setuptools-rust
-  ] ++ (with rustPlatform; [ rust.cargo rust.rustc ]);
+    setuptoolsRustBuildHook
+    rustc
+    cargo
+  ];
 
-  buildInputs = [ openssl ]
+  buildInputs = [ openssl rustPlatform.cargoSetupHook ]
```

```suggestion
  buildInputs = [
  	openssl
  	rustPlatform.cargoSetupHook
  ]
```

I might need a rationally why hooks are now buildInputs as opposed to nativeBuildInputs.

---

Agreed.

@Cynerd, for `rustPlatform.cargoSetupHook` you can probably harvest the explanation from this comment: https://github.com/NixOS/nixpkgs/pull/212795#discussion_r1169379568